rust/xaynet-server/src/state_machine/phases/idle.rs

```diff
@@ -146,6 +135,18 @@ impl PhaseState<Idle> {
     }
 }
 
+#[cfg(test)]
+use crate::state_machine::{phases::Handler, requests::StateMachineRequest, RequestError};
+
+#[cfg(test)]
+#[async_trait]
+impl Handler for PhaseState<Idle> {
```

Comment:
Related to my comment above: why do we need it for the tests? Also maybe this `impl` and the `use` above be moved to the `test` module below?